src/lib/FlightTasks/tasks/ManualAltitude/FlightTaskManualAltitude.cpp

```diff
@@ -289,18 +298,33 @@ void FlightTaskManualAltitude::_rotateIntoHeadingFrame(Vector2f &v)
 
 void FlightTaskManualAltitude::_updateHeadingSetpoints()
 {
-	/* Yaw-lock depends on stick input. If not locked,
-	 * yaw_sp is set to NAN.
-	 * TODO: add yawspeed to get threshold.*/
-	if (fabsf(_yawspeed_setpoint) > FLT_EPSILON) {
-		// no fixed heading when rotating around yaw by stick
-		_yaw_setpoint = NAN;
+	_isYawInput()
+	? _unlockYaw()
+	: _lockYaw();
+}
 
-	} else {
-		// hold the current heading when no more rotation commanded
-		if (!PX4_ISFINITE(_yaw_setpoint)) {
-			_yaw_setpoint = _yaw;
-		}
+bool FlightTaskManualAltitude::_isYawInput()
+{
+	/*
+	 * A threshold larger than FLT_EPSILON is required because the
+	 * _yawspeed_setpoint comes from an IIR filter and takes too much
+	 * time to reach zero.
+	 */
+	return fabsf(_yawspeed_setpoint) > 0.001f;
```

Comment:
A slewrate doesn't need this. Then it's more predictable when it actually locks the heading.

Reply:
Well, I don't really agree with that statement. This gives a constant time (constant multiple of the time constant) between stick release and yaw lock while in case of the slew-rate, the time it takes to lock depends on the amplitude of the motion before releasing the stick.
So it's more predictable for a compute but not for a human.

Reply:
Interesting, the pilot intuivity might be a good point. I was more referring to the fact that I would not be sure how long it takes to reach the lock or is the 10e-3 a predefined value?

Also "the time it takes to lock depends on the amplitude of the motion before releasing the stick" represents more the physical feasibility. If you plan to invest a fixed yaw torque you want to spend to follow the setpoint change then the breaking time depends on the state before starting.